taskcluster/fenix_taskgraph/transforms/multi_dep.py

```diff
@@ -65,19 +67,27 @@ def build_upstream_artifacts(config, tasks):
             "upstream-artifacts": [],
         }
 
-        for dep in _get_all_deps(task).values():
-            paths = sorted([
-                apk_metadata["name"]
-                for apk_metadata in dep.attributes.get("apks", {}).values()
-            ])
-            if paths:
-                worker_definition["upstream-artifacts"].append({
-                    "taskId": {"task-reference": "<{}>".format(dep.kind)},
-                    "taskType": dep.kind,
-                    "paths": paths,
-                })
-
-        task["worker"].update(worker_definition)
+        if "artifact_map" in task["attributes"]:
```

Comment:
Would it maybe be easier to use a separate transform for the beetmover task, instead of repurposing this one?

Reply:
Yeah that could make sense, though technically `beetmover` isn't the only kind that can use "declarative artifacts".. Maybe the util function should be renamed to `generate_declaritive_upstream_artifacts` or something.